cluster/src/main/scala/io/snappydata/gemxd/SparkSQLExecuteImpl.scala

```diff
@@ -237,7 +246,21 @@ class SparkSQLExecuteImpl(val sql: String,
       case t: DecimalType => (StoredFormatIds.SQL_DECIMAL_ID, t.precision, t.scale)
       case FloatType => (StoredFormatIds.SQL_REAL_ID, -1, -1)
       case DoubleType => (StoredFormatIds.SQL_DOUBLE_ID, -1, -1)
-      case StringType => (StoredFormatIds.SQL_CLOB_ID, -1, -1)
+      case s: StringType =>
+        if (snc.sessionState.conf.stringAsClob || stringAsClob) {
+            if (f.metadata.contains(Constant.CHAR_SIZE_NAME) &&
+                !f.metadata.getString(Constant.CHAR_TYPE_NAME).equals("STRING")) {
+              (StoredFormatIds.SQL_VARCHAR_ID,
```

Comment:
I think we need a StoreFormatId mapping for constant width char type also. Before this change all CHAR or VARCHARs were going as CLOB. Now VARCHHAR will go as VARCHAR and even CHARS will go as VARCHARs with this code. Isn't it?